File: jobs/2026-07-03__12-25-47/encore_multi_service_api_ts__RwJvZdg/artifacts/user/myapp/order/order.ts
import { api } from "encore.dev/api";
import { user } from "~encore/clients";

interface OrderResponse {
  orderId: number;
  userId: number;
  userName: string;
}

export const get = api(
  { expose: true, method: "GET", path: "/order/:id" },
  async ({ id }: { id: number }): Promise<OrderResponse> => {
    const u = await user.get({ id: 1 });
    return {
      orderId: id,
      userId: u.id,
      userName: u.name,
    };
  }
);

File: jobs/2026-07-03__05-39-37/encore_multi_service_api_go__yaweQSu/artifacts/user/myproject/order/order.go
// Package order provides order-related APIs.
package order

import (
	"context"

	"myapp-mbx2/user"
)

// OrderResponse is the response returned by the Get API.
type OrderResponse struct {
	ID       string `json:"id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

// Get returns the order with the given id, including the user's details
// fetched via a service-to-service call to the user service.
//
//encore:api path=/order/:id
func Get(ctx context.Context, id string) (*OrderResponse, error) {
	// Look up the order in our static "database".
	o, ok := orders[id]
	if !ok {
		// Default fallback so any id returns an order.
		o = &order{ID: id, UserID: "1"}
	}

	// Service-to-service call to the user service.
	// Encore handles the networking transparently.
	u, err := user.Get(ctx, o.UserID)
	if err != nil {
		return nil, err
	}

	return &OrderResponse{
		ID:       o.ID,
		UserID:   u.ID,
		UserName: u.Name,
	}, nil
}

// order is the internal storage type.
type order struct {
	ID     string
	UserID string
}

// orders is a small static "database" of orders.
var orders = map[string]*order{
	"100": {ID: "100", UserID: "1"},
	"101": {ID: "101", UserID: "2"},
	"102": {ID: "102", UserID: "3"},
}